### PR TITLE
Cleanup: Remove useless UDPConnection function

### PR DESCRIPTION
**Pull request by scw00**
link to #2373

### CODE REVIEW

**Comment by scw00** [comment]
[approve ci FreeBSD]

File: iocore/net/UnixUDPConnection.cc [hunk 1]
```diff
@@ -68,29 +68,20 @@ UnixUDPConnection::callbackHandler(int event, void *data)
     return EVENT_CONT;
   }
 
-  if (m_errno) {
```
Comment (scw00) [inline iocore/net/UnixUDPConnection.cc]: Remove this because nobody call `UnixUDPConnection::errorAndDie `

**Comment by scw00** [comment]
[approve ci CentOS]

File: iocore/net/UnixUDPConnection.cc (hunk 1 above)
Comment (scw00) [reply iocore/net/UnixUDPConnection.cc]: Revert since it was a bug.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

Needed to avoid merge conflicts.

**Pull request merged**